src/node/blockstorage.cpp

```diff
@@ -494,10 +502,11 @@ bool BlockManager::WriteBlockIndexDB()
     return true;
 }
 
-bool BlockManager::LoadBlockIndexDB(const std::optional<uint256>& snapshot_blockhash)
+util::Result<InterruptResult, AbortFailure> BlockManager::LoadBlockIndexDB(const std::optional<uint256>& snapshot_blockhash)
 {
-    if (!LoadBlockIndex(snapshot_blockhash)) {
-        return false;
+    auto result{LoadBlockIndexData(snapshot_blockhash)};
+    if (!result || IsInterrupted(*result)) {
+        return result;
```

Comment:
re: https://github.com/bitcoin/bitcoin/pull/29700#discussion_r1693183422

> Seems fragile to use `util::Result<InterruptResult>`. Wrapping one result into another is fragile, because call-sites may easily forget to unwrap the inner result, especially if the inner result is otherwise `void`. My recommendation would be to treat interruption as `util::Error`, so that callers don't miss it if they check for errors (but forget to check for interruption), but are free to check it, if they inspect the precise error.

It sounds like you want `Result<InterruptResult, void>` to be replaced with `Result<void, InterruptResult>` which would be ok with me. I don't think the difference between these things is very significant, but I agree maybe if the caller forgets to check for the interrupted status, incorrectly treating the interrupted status like a failure might be safer than incorrectly treating the interrupted status like a success.